src/lib/protocols/dns.c

```diff
@@ -513,8 +517,30 @@ static void ndpi_search_dns(struct ndpi_detection_module_struct *ndpi_struct, st
 
   if(flow->packet_counter > 3)
     NDPI_EXCLUDE_PROTO(ndpi_struct, flow);
+    
+  if((flow->packet.detected_protocol_stack[0] == NDPI_PROTOCOL_DNS)
+	 || (flow->packet.detected_protocol_stack[1] == NDPI_PROTOCOL_DNS)) {
+	   
+  	if(flow->packet.udp != NULL && flow->packet.payload_packet_len > PKT_LEN_ALERT)
+  	  ndpi_set_risk(ndpi_struct, flow, NDPI_DNS_LARGE_PACKET);
+  
+  	const struct ndpi_iphdr *iph = flow->packet.iph;
+  	const u_int8_t *l3 = (const u_int8_t *) flow->packet.iph;
+  	const struct ndpi_ipv6hdr *iph_v6 = NULL;
+  	const u_int16_t ipsize = flow->packet.l3_packet_len;
+  	  
+  	if(iph != NULL && iph->version == 6 && ipsize >= sizeof(struct ndpi_ipv6hdr)) {
+  	  iph_v6 = (const struct ndpi_ipv6hdr *) l3;
+  	  iph = NULL;
+  	}
+  	  
+  	if((iph != NULL && (ipsize < iph->ihl * 4 || ipsize < ntohs(iph->tot_len) || ntohs(iph->tot_len) < iph->ihl * 4 
```

Comment:
Could you add another comment here about ndpiReader?
Something like "Note that ndpiReader drops any IP-fragmented packets, so this check will be always false when using it"

Reply:
I'm editing the dns.c file like you suggested, but I don't understand this review.
Are you saying that the whole 'if' (from 537) always returns false? In this case, there must be something off because using the pcap file downloded here: https://weberblog.net/dns-capture-udp-tcp-ip-fragmentation-edns-ecs-cookie/ I see the flow risk about fragmented DNS packets.

Reply:
You are right, sorry for the noise. 
ndpiReader drops only ip-fragmented packets with ip-offset != 0, i.e. only the first fragment is passed to the nDPI code.
Ignore this comment; please add the other one about EDNS, please